### PR TITLE
Transfer VTK protocols and serializers to trame-vtk

### PR DESCRIPTION
**Pull request by psavery**
In this PR, everything that was previously being used in the `vtkmodules.web` directory [in vtk](https://gitlab.kitware.com/vtk/vtk/-/tree/519316fe11817a95965f9c231b5ef5f85dc8490c/Web/Python/vtkmodules/web) has been transferred to `trame-vtk`.

A couple of additional changes have been made as well:

1. The `protocols` file was broken up into separate files for each class. This was done in [this commit](https://github.com/Kitware/trame-vtk/commit/671544ead923fc27d86fa94ca05bfab56292033f).
2. The `addon_serializer.py` file has been deleted and all patches applied to the `render_window_serializer.py` file were transferred over to it. Those changes can be seen in [this commit](https://github.com/Kitware/trame-vtk/commit/5679ddb27113c896fab73ab61364aa2c4edf17ae) (note that the commit message includes all potential discrepancies I found).

To do:
 - [x] Transfer protocols from `paraview.web`
 - [x] Perform thorough testing using the examples
 - [x] Break apart serializers into separate files and simplify further
 - [x] Integrate last action check into VTK mouse handler (take it from ParaView)
 - [x] Add environment variable for printing serialization errors
 - [x] Fix paraview mouse wheel scroll to be similar to the VTK one

Validation tests confirmed working:
- [x] AddRemoveActors
- [x] AxesActor
- [x] CaptureImageLocal
- [x] CaptureImage
- [x] CaptureImageRemote
- [x] ColorByComponent
- [x] ManyViews
- [x] NoSize
- [x] PushCamera
- [x] PyVistaAxesGrid
- [x] PyVistaFirstStillRatio
- [x] PyVistaInt64
- [x] PyVistaLookupTable
- [x] SwitchView

### CODE REVIEW

File: trame_vtk/modules/vtk/protocols/file_browser.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import os
```
Comment (jourdain) [inline trame_vtk/modules/vtk/protocols/file_browser.py:1]: Are we using that protocol?

Comment (psavery) [reply trame_vtk/modules/vtk/protocols/file_browser.py:1]: Doesn't look like it. I just just transferred them all over. Do you want me to delete the unused ones?

File: trame_vtk/modules/vtk/protocols/view_port_geometry_delivery.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from wslink import register as exportRpc
```
Comment (jourdain) [inline trame_vtk/modules/vtk/protocols/view_port_geometry_delivery.py:1]: We should probably remove that one

File: trame_vtk/modules/vtk/protocols/file_browser.py (hunk 1 above)
Comment (jourdain) [reply trame_vtk/modules/vtk/protocols/file_browser.py:1]: yes, so we only have the piece of code we really need to worry about

Comment (psavery) [reply trame_vtk/modules/vtk/protocols/file_browser.py]: Done.

File: trame_vtk/modules/vtk/protocols/view_port_image_delivery.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import time
+
+from wslink import register as exportRpc
+
+from .web_protocol import vtkWebProtocol
+
+
+class vtkWebViewPortImageDelivery(vtkWebProtocol):
```
[the remark this answers is not in the document]
Comment (psavery) [reply trame_vtk/modules/vtk/protocols/view_port_image_delivery.py]: Done.

File: trame_vtk/modules/paraview/protocols/local_rendering.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
+from paraview import simple
+from wslink import register as exportRpc
+
+from trame_vtk.modules.vtk.protocols.serializers import (
```
Comment (jourdain) [inline trame_vtk/modules/paraview/protocols/local_rendering.py:4]: serializers are independent of protocols... They should be siblings.

File: trame_vtk/modules/paraview/protocols/publish_image_delivery.py [hunk 5]
```diff
@@ -0,0 +1,630 @@
...
+        viewProxy = self.getView(event["view"])
+        if viewProxy and "spinY" in event:
+            rootId = viewProxy.GetGlobalIDAsString()
+            zoomFactor = 1.0 - event["spinY"] / 10.0
+
+            if rootId in self.linkedViews:
+                fp = viewProxy.CameraFocalPoint
+                pos = viewProxy.CameraPosition
+                delta = [fp[i] - pos[i] for i in range(3)]
+                viewProxy.GetActiveCamera().Zoom(zoomFactor)
```
Comment (jourdain) [inline trame_vtk/modules/paraview/protocols/publish_image_delivery.py:594]: This probably will need to do the same thing you do on the VTK side...

File: trame_vtk/modules/vtk/__init__.py [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import warnings
 
 from .core import HybridView
+from .protocols.serializers import mesh as vtk_mesh
```
Comment (jourdain) [inline trame_vtk/modules/vtk/__init__.py:4]: Should be `from .serializers import ...`

File: trame_vtk/modules/vtk/protocols/serializers/__init__.py
Comment (jourdain) [inline trame_vtk/modules/vtk/protocols/serializers/__init__.py:1]: Move them as sibling of protocols

File: trame_vtk/modules/vtk/protocols/serializers/pwf.py
Comment (jourdain) [inline trame_vtk/modules/vtk/protocols/serializers/pwf.py:1]: could be in lookup_tables?

**Review by jourdain — COMMENTED** [review]
Looks good with minor file move...

File: trame_vtk/modules/vtk/protocols/serializers/__init__.py
Comment (psavery) [reply trame_vtk/modules/vtk/protocols/serializers/__init__.py:1]: Done.

File: trame_vtk/modules/vtk/__init__.py (hunk 6 above)
Comment (psavery) [reply trame_vtk/modules/vtk/__init__.py]: Done

File: trame_vtk/modules/paraview/protocols/local_rendering.py (hunk 4 above)
Comment (psavery) [reply trame_vtk/modules/paraview/protocols/local_rendering.py]: Done.

File: trame_vtk/modules/vtk/protocols/serializers/pwf.py
Comment (psavery) [reply trame_vtk/modules/vtk/protocols/serializers/pwf.py:1]: Done.

File: trame_vtk/modules/vtk/serializers/registry.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
+SERIALIZERS = {}
+JS_CLASS_MAPPING = {}
+context = None
```
Comment (jourdain) [inline trame_vtk/modules/vtk/serializers/registry.py:3]: is `context` needed here?

**Review by jourdain — APPROVED** [review]

**Review by banesullivan — APPROVED** [review]
At this point all looks good from the PyVista side! Thanks for all this hard work, @psavery! Going to make maintaining the serializer and protocols much, much more streamlined

**Review by jourdain — APPROVED** [review]

File: trame_vtk/modules/vtk/serializers/registry.py (hunk 7 above)
Comment (psavery) [reply trame_vtk/modules/vtk/serializers/registry.py]: It is now removed.

**Pull request merged**